internal/cmd/test.go

```diff
@@ -138,3 +139,20 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, expectedExitCode int, index int) (output string, errs error) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		errs = errors.Join(errs, fmt.Errorf("Case %d: CMD run returned an error: '%v'.", index, err))
+	}
+
+	if len(stderr) > 0 {
+		errs = errors.Join(errs, fmt.Errorf("Case %d: CMD has content in stderr: '%s'.", index, stderr))
+	}
+
+	if expectedExitCode != exitCode {
+		errs = errors.Join(errs, fmt.Errorf("Case %d: Unexpected exit code. Expected: '%d', Actual: '%d'.", index, expectedExitCode, exitCode))
+	}
+
+	return stdout, errs
```

Comment:
I would just return `stdout, stderr, exitCode`.
Most people will ignore all returns from this function,
but we can include all just in case.